File: day1/day1.py
from helpers import parse_input_into_two_lists

list1, list2 = parse_input_into_two_lists('real_input1.txt')
def part1():
    total = 0
    list1.sort()
    list2.sort()
    for i in range(len(list1)):
        total += abs(list1[i]-list2[i])

    return total

def part2():
    total = 0
    for item in list1:
        total += item * list2.count(item)

    return total

print(part1())
print(part2())

File: day1/day1.clj
(ns day1)

(defn parse_input [filename]
  (as-> filename input
  (clojure.string/split (slurp input) #"\n")
  (map (fn [x] (clojure.string/split x #"   ")) input)
  (map (fn [x] (map #(Integer/parseInt %) x)) input)
  (map (fn [x] (vec x)) input))
  )

(defn two_lists [input]
  (let [l1 (vec (map first input))
        l2 (vec (map second input))] 
    [(sort l1) (sort l2)]
    )
  )

(defn list_diff [[l1 l2]]
  (map (comp abs -) l1 l2)
  )

(defn single_score [digit list]
  (* digit (count (filter #(= digit %) list))))

(defn similarity_score [[list1 list2]]
  (reduce + (map #(single_score % list2) list1))
  )

(defn part1 [filename]
  (->> filename
       parse_input
       two_lists
       list_diff
       (reduce +)
       )
  )

(defn part2 [filename]
  (->> filename
       parse_input
       two_lists
       similarity_score
       )
  )

(part1 "day1/real_input1.txt")
(part2 "day1/real_input1.txt")